src/QirRuntime/lib/QSharpCore/.clang-tidy

```diff
@@ -0,0 +1 @@
+Checks: '-*,bugprone-*'
```

Comment:
I am wondering what were the specific reasons for not using the parent's clang-tidy configuration?

One pattern that might be useful if it is just a couple of checks that are different from the parent configuration would be using `InheritParentConfig: true` and adding/removing the necessary checks.